### ISSUE TITLE
[Windows] 1s pause on empty window after opening file from command line

### ISSUE THREAD

**Issue by crabmusket** [trimental/inlyne#249]
**Platform:** 

```
Processor	Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz   2.59 GHz
Installed RAM	32.0 GB (31.8 GB usable)
Edition		Windows 11 Pro
Version		22H2
OS build	22621.3155
Experience	Windows Feature Experience Pack 1000.22684.1000.0
```

**Installation method:**

```
> cargo install inlyne
> inlyne --version
inlyne 0.4.0
```

**Issue:**

When opening even a tiny markdown file from PowerShell, there is a 1-1.5s pause on a blank window which gives a "busy" spinning cursor.

https://github.com/trimental/inlyne/assets/904269/d5164225-1fea-4aa6-bab8-da8fdc937ca0

I'm happy to try to find out more information when I have time. I'm not used to debugging software on Windows (I do most of my work in WSL) so would love suggestions!

**Comment by CosmicHorrorDev** [comment]
Thanks for opening an issue!

Can you try explicitly setting the font families in [`inlyne`'s config file](https://github.com/Inlyne-Project/inlyne/tree/v0.4.1?tab=readme-ov-file#configuration) to see if that speeds things up at all?

```toml
[font-options]
regular-font = "..."
monospace-font = "..."
```

**Comment by crabmusket** [comment]
Added this to `config.toml`:

```toml
[font-options]
regular-font = "Arial"
monospace-font = "Consolas"
```

And running `inlyne -c config.toml title.md` I don't see any speedup.

I'm going to try compiling from source and doing some good old print debugging this weekend!

**Comment by crabmusket** [comment]
I've done some print-profiling and it seems that this block is the one that takes all the time. Everything else seems pretty much instant.

https://github.com/Inlyne-Project/inlyne/blob/d13402867d92147314937b182dcbcbabe7c94a23/src/main.rs#L188-L194

**Comment by CosmicHorrorDev** [comment]
Thanks for helping narrow things down. Unfortunately I'm not sure what the best thing for us to do is here :sweat_smile: 

You can try reporting things upstream to wgpu or asking in one of their help channels, since they'll be way more equipped to handle these kinds of things

**Comment by crabmusket** [comment]
Yep, will do and report back. If that Reddit reply is accurate it may be out of your hands entirely.

I guess maybe the only possible thing to do would be to render a loading indicator in the window, so it's not just a busy cursor? But that would probably involve rearranging the code a fair bit for a very small aesthetic improvement on a single platform.
